server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java

```diff
@@ -126,11 +142,29 @@ public List<Table> list(@NonNull @Valid QueryTableParams params) throws SQLExcep
                         tableExtension);
             }
             if (types.contains(DBObjectType.EXTERNAL_TABLE)) {
-                generateListAndSyncDBTablesByTableType(params, database, dataSource, tables, conn,
-                        DBObjectType.EXTERNAL_TABLE, tableExtension);
+                if (checkExternalTableSupported(dataSource.getDialectType(), conn)) {
+                    generateListAndSyncDBTablesByTableType(params, database, dataSource, tables, conn,
+                            DBObjectType.EXTERNAL_TABLE, tableExtension);
+                }
             }
-            return tables;
         }
+        return tables;
+    }
+
+    @SkipAuthorize("permission check inside")
+    public boolean checkExternalTableSupported(@NonNull DialectType dialectType, @NonNull Connection conn)
+            throws SQLException {
+        String databaseProductVersion = conn.getMetaData().getDatabaseProductVersion();
```

Comment:
```java
 InformationExtensionPoint point =
                ConnectionPluginUtil.getInformationExtension(connectionSession.getDialectType());
        point.getDBVersion()
```
these codes may help you get the version

Reply:
ok